### PR TITLE
Feature/limit value lengths

### PR DESCRIPTION
**Pull request by frascuchon**
Server side:

- Truncate metadata string values 
- entity name limit validation
- label  limit validation

Client size:

- Truncate metadata string values with single warning

### CODE REVIEW

**Review by ignacioct — APPROVED** [review]
LGTM. Good work!

File: src/rubrix/server/tasks/text_classification/api/model.py [hunk 1]
```diff
@@ -32,6 +33,16 @@ class ClassPrediction(BaseModel):
     class_label: Union[str, int] = Field(alias="class")
     score: float = Field(default=1.0, ge=0.0, le=1.0, alias="confidence")
 
+    @validator("class_label")
+    def check_label_length(cls, class_label):
+        if isinstance(class_label, str):
+            assert 1 <= len(class_label) <= MAX_KEYWORD_LENGTH, (
+                f"Class {class_label} exceeds max length"
```
Comment (dcfidalgo) [inline src/rubrix/server/tasks/text_classification/api/model.py]:
```suggestion
                f"Class name '{class_label}' exceeds max length of {MAX_KEYWORD_LENGTH}"
```

File: src/rubrix/server/tasks/text_classification/api/model.py [hunk 2]
```diff
@@ -32,6 +33,16 @@ class ClassPrediction(BaseModel):
...
+                if len(class_label) > MAX_KEYWORD_LENGTH
+                else f"Class must not be empty"
```
Comment (dcfidalgo) [inline src/rubrix/server/tasks/text_classification/api/model.py]:
```suggestion
                else f"Class name must not be empty"
```

File: src/rubrix/server/commons/helpers.py [hunk 3]
```diff
@@ -67,3 +67,36 @@ def unflatten_dict(
...
+    max_length:
+        Max length for string values
+
+    Returns
+    -------
+        Limited version of data, if any
+    """
+
+    if isinstance(data, str):
+        return data[len(data) - max_length :]
```
Comment (dcfidalgo) [inline src/rubrix/server/commons/helpers.py]:
```suggestion
        return data[-max_length:]
```

File: src/rubrix/server/commons/helpers.py [hunk 4]
```diff
@@ -67,3 +67,36 @@ def unflatten_dict(
...
+        return {
+            k: limit_value_length(v, max_length=max_length) for k, v in data.items()
+        }
+    if isinstance(data, (list, tuple, set)):
+        new_values = [limit_value_length(v, max_length=max_length) for v in data]
+        if isinstance(data, tuple):
+            return tuple(new_values)
+        if isinstance(data, set):
+            return set(new_values)
+        return new_values
```
Comment (dcfidalgo) [inline src/rubrix/server/commons/helpers.py]: perhaps a bit more elegant
```suggestion
        new_values = map(lambda x: limit_value_length(x, max_length=max_length), data)
        return type(data)(new_values)
```

File: src/rubrix/client/models.py [hunk 5]
```diff
@@ -58,6 +61,17 @@ class TokenAttributions(BaseModel):
     attributions: Dict[str, float] = Field(default_factory=dict)
 
 
+def limit_metadata_values(metadata: Dict[str, Any]) -> Dict[str, Any]:
+    """Checks metadata values length and apply value truncation for large values"""
+    new_value = limit_value_length(metadata, max_length=MAX_KEYWORD_LENGTH)
+    if new_value != metadata:
+        warnings.warn(
+            "Some metadata values exceeds max length"
+            f"Those values will be truncated keeping only last {MAX_KEYWORD_LENGTH} characters."
```
Comment (dcfidalgo) [inline src/rubrix/client/models.py]:
```suggestion
            "Some metadata values exceed the max length. "
            f"Those values will be truncated by keeping only the last {MAX_KEYWORD_LENGTH} characters."
```

**Review by dcfidalgo — APPROVED** [review]
Nice! Just a few inline chorradas

File: src/rubrix/server/commons/helpers.py (hunk 4 above)
Comment (frascuchon) [reply src/rubrix/server/commons/helpers.py]: ¿list(_list) == _list?

Comment (frascuchon) [reply src/rubrix/server/commons/helpers.py]: it looks like it is.

Comment (dcfidalgo) [reply src/rubrix/server/commons/helpers.py]: hm, if you are referring to the return type of map, map returns an iterable, yielding the results.

**Pull request merged**